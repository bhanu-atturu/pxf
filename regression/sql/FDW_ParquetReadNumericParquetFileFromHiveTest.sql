-- start_ignore
{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} mkdir -p {{ TEST_LOCATION }}
{{ GPDB_REMOTE }}-- if GPDB is remote, will need to scp file down from there for beeline
{{ GPDB_REMOTE }}\!scp {{ PGHOST }}:{{ TEST_LOCATION }}/data.csv {{ TEST_LOCATION }}
\!{{ HCFS_CMD }} --config {{ PXF_BASE }}/servers/{{ SERVER_CONFIG }}/ dfs -mkdir -p '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/parquet/'
\!{{ HCFS_CMD }} --config {{ PXF_BASE }}/servers/{{ SERVER_CONFIG }}/ dfs -copyFromLocal '{{ WORKING_DIR }}/resources/data/parquet/numeric.parquet' '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/parquet/numeric.parquet'
-- end_ignore

CREATE SERVER parquet_test_parquet_read_numeric
    FOREIGN DATA WRAPPER {{ HCFS_PROTOCOL }}_pxf_fdw
    OPTIONS (config '{{ SERVER_CONFIG }}');

CREATE USER MAPPING FOR CURRENT_USER SERVER parquet_test_parquet_read_numeric;

DROP FOREIGN TABLE IF EXISTS pxf_parquet_read_numeric CASCADE;

CREATE FOREIGN TABLE pxf_parquet_read_numeric
    (
        description TEXT,
        a DECIMAL(5, 2),
        b DECIMAL(12, 2),
        c DECIMAL(18, 18),
        d DECIMAL(24, 16),
        e DECIMAL(30, 5),
        f DECIMAL(34, 30),
        g DECIMAL(38, 10),
        h DECIMAL(38, 38)
        ) SERVER parquet_test_parquet_read_numeric
    OPTIONS (resource '{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/parquet/numeric.parquet', format 'parquet');

-- @description test numeric with precision and scale defined
SELECT a, b, c, d, e, f, g, h FROM pxf_parquet_read_numeric ORDER BY description;

-- start_ignore
{{ CLEAN_UP }}-- clean up HCFS and local disk
{{ CLEAN_UP }} DROP FOREIGN TABLE IF EXISTS pxf_parquet_read_numeric CASCADE;
{{ CLEAN_UP }} DROP USER MAPPING IF EXISTS FOR CURRENT_USER SERVER parquet_test_parquet_read_numeric;
{{ CLEAN_UP }} DROP SERVER IF EXISTS parquet_test_parquet_read_numeric CASCADE;
{{ CLEAN_UP }}\!{{ HCFS_CMD }} --config {{ PXF_BASE }}/servers/{{ SERVER_CONFIG }}/ dfs -rm -r -f {{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/parquet/numeric.parquet
{{ CLEAN_UP }}{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} rm -rf {{ TEST_LOCATION }}
-- end_ignore