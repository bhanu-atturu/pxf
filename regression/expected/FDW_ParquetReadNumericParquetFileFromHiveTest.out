-- start_ignore
-- end_ignore

CREATE SERVER parquet_test_parquet_read_numeric
    FOREIGN DATA WRAPPER {{ HCFS_PROTOCOL }}_pxf_fdw
    OPTIONS (config '{{ SERVER_CONFIG }}');

CREATE USER MAPPING FOR CURRENT_USER SERVER parquet_test_parquet_read_numeric;

DROP FOREIGN TABLE IF EXISTS pxf_parquet_read_numeric CASCADE;
NOTICE:  foreign table "pxf_parquet_read_numeric" does not exist, skipping

CREATE FOREIGN TABLE pxf_parquet_read_numeric
    (
        description TEXT,
        a DECIMAL(5, 2),
        b DECIMAL(12, 2),
        c DECIMAL(18, 18),
        d DECIMAL(24, 16),
        e DECIMAL(30, 5),
        f DECIMAL(34, 30),
        g DECIMAL(38, 10),
        h DECIMAL(38, 38)
        ) SERVER parquet_test_parquet_read_numeric
    OPTIONS (resource '{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/parquet/numeric.parquet', format 'parquet');

-- @description test numeric with precision and scale defined
SELECT a, b, c, d, e, f, g, h FROM pxf_parquet_read_numeric ORDER BY description;
    a    |       b        |           c           |             d              |                e                 |                  f                   |                    g                     |                     h                     
---------+----------------+-----------------------+----------------------------+----------------------------------+--------------------------------------+------------------------------------------+-------------------------------------------
    0.00 |  1234567890.12 |  0.123456789012345678 |  95162384.7951623847951623 |  9876543210987654321098765.43210 |  1592.634871592634871592634871592634 |  7410258963852014796374102589.6328931478 |  0.45612398707410963258015926348791374682
  123.45 |           0.00 | -0.123456789012345678 | -95162384.7951623847951623 | -9876543210987654321098765.43210 | -1592.634871592634871592634871592634 | -7410258963852014796374102589.6328931478 | -0.45612398707410963258015926348791374682
 -123.45 | -1234567890.12 |  0.000000000000000000 |  10035779.8012365012872940 |  1054236548795004545620498.75065 |  1111.111111111111111111111111111111 |  9000000000000000000000000000.0000000009 |  0.11111111111111111111111111111111111111
    0.25 |           0.25 |  0.252525252525252525 |         0.0000000000000000 | -1054236548795004545620498.75065 | -1111.111111111111111111111111111111 | -9000000000000000000000000000.0000000009 | -0.11111111111111111111111111111111111111
   -0.25 |          -0.25 | -0.252525252525252525 | -10035779.8012365012872940 |                          0.00000 |     0.252525252525252525252525252525 |                             0.2525252525 |  0.25252525252525252525252525252525252525
    0.99 |  9223372036.00 |  0.984195165432541981 |  87922089.0000000000000000 |                         15.00000 |     0.000000000000000000000000000000 |                            -0.2525252525 | -0.25252525252525252525252525252525252525
   -0.99 | -9223372036.00 | -0.984195165432541981 | -87922089.0000000000000000 |                        -15.00000 |    26.000000000000000000000000000000 |                             0.0000000000 |  0.10000000000000000000000000000000000000
  500.00 |  1000000000.00 |  0.500000000000000000 |         1.0055129872001568 |  2515950205015098070500051.00000 |   -26.000000000000000000000000000000 |  2815602548981565158984895010.0000000000 |  0.00000000000000000000000000000000000000
(8 rows)

-- start_ignore
-- end_ignore